File: archive/p/python/duplicate_char_counter.py
import sys
import logging


def duplicate_counter(text):
    char_dict = {}  # Dictionary where we store the characters and their occurrence amount
    for i in text:
        if text.count(i) > 1 and i not in char_dict:  # If i occurs more than once and hasn't already been done
            char_dict.update({i:text.count(i)})  # Add the key and value to char_dict{}
    return char_dict


def main(args):
    try:
        for char, occurrences in duplicate_counter(args[0]).items():  # Iterate through arguments's letters and find their respective occurrence amount
            print("Characters: {}, Occurrences: {}".format(char, occurrences))  # Print it to console, with easy to read formatting
    except IndexError:
        print("Please provide a valid string.")
    except Exception as e:
        logging.exception(e)  # Log any exceptions


if __name__ == "__main__":
    main(sys.argv[1:])

File: archive/j/javascript/duplicate-char-counter.js
const string = process.argv[2];
if (!string) return console.log("Please provide a string.");

chars_counter = {};

for (const char of string) {
    if (!chars_counter[char]) chars_counter[char] = 0;
    chars_counter[char] += 1;
}

for (const char_count of Object.entries(chars_counter)) {
    if (char_count[1] >= 2) {
        console.log(
            `Character: ${char_count[0]}, Occurrences: ${char_count[1]}`
        );
    }
}
